pkg/controllers/provisioning/scheduling/topologynodefilter.go

```diff
@@ -0,0 +1,69 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
+)
+
+// TopologyNodeFilter is used to determine if a given actual node or scheduling node matches the pod's node selectors
+// and required node affinity terms.  This is used with topology spread constraints to determine if the node should be
+// included for topology counting purposes. This is only used with topology spread constraints as affinities/anti-affinities
+// always count across all nodes. A nil or zero-value TopologyNodeFilter behaves well and the filter returns true for
+// all nodes.
+type TopologyNodeFilter []v1alpha5.Requirements
```

Comment:
Now that this type has been simplified, it occurs to me that it might be more straightforward to fold this into topologygroup e.g. `nodeFilter []v1alpha5.Requirements`. It should be ~40 lines of code.